src/geometry/broad_phase_multi_sap/broad_phase_multi_sap.rs

```diff
@@ -231,7 +231,12 @@ impl BroadPhaseMultiSap {
          * Actually remove the colliders proxies.
          */
         for removed in removed_colliders {
-            if let Some(proxy_id) = self.colliders_proxy_ids.remove(removed) {
+            #[cfg(feature = "enhanced-determinism")]
+            let proxy_id = self.colliders_proxy_ids.shift_remove(removed);
+            #[cfg(not(feature = "enhanced-determinism"))]
+            let proxy_id = self.colliders_proxy_ids.remove(removed);
```

Comment:
These are modified because of a deprecated warning:

```
warning: use of deprecated method `indexmap::map::IndexMap::<K, V, S>::remove`: `remove` disrupts the map order -- use `swap_remove` or `shift_remove` for explicit behavior.
   --> crates/rapier3d/../../src/geometry/broad_phase_multi_sap/broad_phase_multi_sap.rs:234:62
    |
234 |             if let Some(proxy_id) = self.colliders_proxy_ids.remove(removed) {
    |                                                              ^^^^^^
    |
    = note: `#[warn(deprecated)]` on by default
```